### ISSUE TITLE
SVG rendering does not use fonts registered with `FontManager`

### ISSUE THREAD

**Issue by ebarnard** [QuestPDF/QuestPDF#937]
**Describe the bug**
When rendering SVGs, fonts registered via `FontManager` do not appear to be used to render text in the SVGs.

**To Reproduce**
- Ensure that the font being registered is *not* in the output directory, so it not loaded automatically by Skia
- Set `QuestPDF.Settings.UseEnvironmentFonts = false;`
- Register a font that is not Lato using `FontManager`
- Render an SVG containing text in the newly registered font

**Expected behavior**
Text is rendered in the registered font

**Screenshots**
![image](https://github.com/QuestPDF/QuestPDF/assets/1059683/61884734-3f54-4206-90e1-9a059f0e7091)

**Environment**
QuestPDF 2024.6.4
Windows 10 x64

**Comment by MarcinZiabek** [comment]
Indeed, text in SVG images is a little bit different. How do you distribute your font files? Are they included as EmbeddedResource?

For SVG to work, you may need to publish your font files within the application directory (e.g. `bin` folder). The library will automatically read all font files published alongside your application.

**Comment by ebarnard** [comment]
I was using embedded resources. Copying to the output/publish directory does fix it. It's not ideal but definitely workable.

It looks like SVG font selection exclusively uses the provided `SkFontMgr` (created by `SkFontMgr_New_Custom_Directory` in the case of not using system fonts). In order to get fonts registered with `FontManager` to work with SVGs it would be necessary to create our own wrapper implementation of the `SkFontMgr` base class which first searches for fonts registered with `FontManager` and then falls back to another `SkFontMgr` (`SkFontMgr_New_Custom_Directory`/the font manager for system fonts).

This sounds like quite a lot of work, so probably not worth it given the available workaround. It would be good to document this limitation of `FontManager` (if it isn't already and I just missed it).

**Comment by MarcinZiabek** [comment]
I completely agree! I don't think it is a common use case. As you have suggested, it is most likely fine as long as the documentation is accurate. For now, I have updated the documentation (https://www.questpdf.com/api-reference/image.html#svg-support). We can keep this enhancement for the future 😄

**Comment by M0ns1gn0r** [comment]
I find the documentation to be not clear enough on the following: no matter what `QuestPDF.Settings.UseEnvironmentFonts` is set to, you **MUST** include the fonts into the application folder. Otherwise the fonts will work for "standard" PDF layout but will suddenly not be found in SVGs.

**Comment by DawidRobertNowak** [comment]
Hello,

Thank you all for your patience with this one, and for the detailed analysis in this thread.

As of version 2026.9.0, SVG text uses fonts registered through `FontManager`, including fonts loaded from streams and embedded resources. Publishing the font files to the application directory is no longer required, and font family names are matched case-insensitively. SVG text also accepts comma-separated fallback lists, such as `font-family="'Open Sans', Lato"`.

Two things worth knowing before upgrading:

- Fonts installed in the operating system are used in SVG only when `Settings.UseSystemFonts` is enabled, which is now disabled by default.
- `Settings.ThrowOnMissingFontFamilies` and `Settings.ThrowOnMissingTextGlyphs` do not apply to SVG content, so an unavailable font in an SVG image does not raise an exception. Please make sure the families referenced by your SVG files are registered or included in a fallback list.

The documentation has been updated accordingly: https://www.questpdf.com/api-reference/image/svg.html#font-support

We are therefore closing this issue as resolved. If the behavior still does not match your setup, please don't hesitate to open a new issue.

Thank you!
